src/map/elements/crab.rs

```diff
@@ -0,0 +1,204 @@
+use super::*;
+use rand::Rng;
+
+pub struct CrabDecorationPlugin;
+
+impl Plugin for CrabDecorationPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_rollback_system(RollbackStage::PreUpdate, hydrate_crab)
+            .add_rollback_system(RollbackStage::Update, update_crabs);
```

Comment:
Here we also need to register the `CrabCritter` component with rollback plugin so it can be snapshot/restored in network games.

```suggestion
            .add_rollback_system(RollbackStage::Update, update_crabs)
            .extend_rollback_plugin(|plugin| {
                plugin
                    .register_rollback_type::<CrabCritter>()
            });
```

You will also need to derive `Reflect` on `CrabCritter` for that to work.

Reply:
I can't derive `Reflect` on `CrabCritter`, since `Reflect` can't be derived on `CrabState`. This is because `CrabState` is an enum. Should I represent the `crab_state` as a byte in the `CrabCritter` struct? @zicklag

Reply:
Try this to derive reflect on the enum:

```rust
#[derive(Reflect, Clone, Serialize, Deserialize)]
#[reflect_value(Serialize, Deserialize)]
pub enum Example {
    ExampleVariant,
}
```